File: 250518/두 방향 탈출 가능 여부 판별하기/determine-escapableness-with-2-ways.cpp
#include <iostream>
using namespace std;
int n, m;
int grid[110][110];
bool visited[110][110];
void search(int x, int y) {
    if(x == n - 1 && y == m - 1) {
        printf("1");
        exit(0);
    }
    if(grid[x + 1][y] && !visited[x + 1][y]) search(x + 1, y);
    if(grid[x][y + 1] && !visited[x][y + 1]) search(x, y + 1);
    visited[x][y] = true;
}
int main() {
    cin >> n >> m;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            cin >> grid[i][j];
        }
    }
    search(0, 0);
    printf("0");
    return 0;
}

File: 250611/두 방향 탈출 가능 여부 판별하기/determine-escapableness-with-2-ways.py
n, m = map(int, input().split())
grid = [list(map(int, input().split())) for _ in range(n)]

def search(x, y):
    if x == n - 1 and y == m - 1:
        print(1)
        exit(0)
    if 0 <= x < n and 0 <= y + 1 < m and grid[x][y + 1]: search(x, y + 1)
    if 0 <= x + 1 < n and 0 <= y < m and grid[x + 1][y]: search(x + 1, y)

search(0, 0)
print(0)